CONTRIBUTING.md

```diff
@@ -0,0 +1,94 @@
+
+- [Your First Contribution](#your-first-contribution)
+- [Workflow](#workflow)
+  - [Git Branches](#git-branches)
+  - [GitHub Pull Requests](#github-pull-requests)
+    - [Title](#title)
+    - [Reviews & Approvals](#reviews--approvals)
+    - [Merge Style](#merge-style)
+    - [CI](#ci)
+- [Setup](#setup)
+  - [Using a dev container environment](#using-a-dev-container-environment)
+  - [Bring your own toolbox](#bring-your-own-toolbox)
+- [Code of Conduct](#code-of-conduct)
+
+## Your First Contribution
+
+1. [Fork the iceberg-rust repository](https://github.com/apache/iceberg-rust/fork) in your own GitHub account.
+1. [Create a new Git branch](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-and-deleting-branches-within-your-repository).
+1. Make your changes.
+1. [Submit the branch as a pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork) to the main iceberg-rust repo. An iceberg-rust team member should comment and/or review your pull request within a few days. Although, depending on the circumstances, it may take longer.
+
+## Workflow
+
+### Git Branches
+
+*All* changes must be made in a branch and submitted as [pull requests](#github-pull-requests). iceberg-rust does not adopt any type of branch naming style, but please use something descriptive of your changes.
+
+### GitHub Pull Requests
+
+Once your changes are ready you must submit your branch as a [pull request](https://github.com/apache/iceberg-rust/pulls).
+
+#### Title
+
+The pull request title must follow the format outlined in the [conventional commits spec](https://www.conventionalcommits.org). [Conventional commits](https://www.conventionalcommits.org) is a standardized format for commit messages. iceberg-rust only requires this format for commits on the `main` branch. And because iceberg-rust squashes commits before merging branches, this means that only the pull request title must conform to this format.
+
+The following are all good examples of pull request titles:
+
+```text
+feat(schema): Add last_updated_ms in schema
+docs: add hdfs classpath related troubleshoot
+ci: Mark job as skipped if owner is not apache
+fix(schema): Ignore prefix if it's empty
+refactor: Polish the implementation of read parquet
+```
+
+#### Reviews & Approvals
+
+All pull requests should be reviewed by at least one iceberg-rust committer.
+
+#### Merge Style
+
+All pull requests are squash merged. We generally discourage large pull requests that are over 300-500 lines of diff. If you would like to propose a change that is larger we suggest coming onto our [mail list](mailto:dev@iceberg.apache.org) and discuss it with us. This way we can talk through the solution and discuss if a change that large is even needed! This will produce a quicker response to the change and likely produce code that aligns better with our process.
```

Comment:
```suggestion
All pull requests are squash merged. We generally discourage large pull requests that are over 300-500 lines of diff. If you would like to propose a change that is larger we suggest coming onto [Iceberg's DEV mailing list](mailto:dev@iceberg.apache.org) and discuss it with us. This way we can talk through the solution and discuss if a change that large is even needed! This will produce a quicker response to the change and likely produce code that aligns better with our process.
```

Reply:
I would probably mention the #rust Slack channel here as well, because some people prefer Slack over Mail

Reply:
Added.